src/pages/groupComparison/MultipleCategoryBarPlot.tsx

```diff
@@ -424,12 +435,57 @@ export default class MultipleCategoryBarPlot extends React.Component<
     }
 
     @computed get labels() {
+        _.forEach(this.data, item => {
+            // Sorting counts within each item
+            item.counts.sort((a, b) =>
+                a.majorCategory.localeCompare(b.majorCategory)
+            );
+        });
+        const totalSumArray: TotalSumItem[] = [];
+        _.forEach(this.data, item => {
+            _.forEach(item.counts, countItem => {
+                const existingItem = _.find(
+                    totalSumArray,
+                    sumItem => sumItem.majorCategory === countItem.majorCategory
+                );
+                if (existingItem) {
+                    existingItem.sum += countItem.count;
+                    existingItem.minorCategory.push({
+                        name: item.minorCategory,
+                        count: countItem.count,
+                        percentage: countItem.percentage,
+                    });
+                } else {
+                    totalSumArray.push({
+                        majorCategory: countItem.majorCategory,
+                        sum: countItem.count,
+                        minorCategory: [
+                            {
+                                name: item.minorCategory,
+                                count: countItem.count,
+                                percentage: countItem.percentage,
+                            },
+                        ],
+                    });
+                }
+            });
+        });
+
+        totalSumArray.sort((a, b) => b.sum - a.sum);
+
+        const sortedLabels = totalSumArray.map(item => item.majorCategory);
+
+        if (this.props.sortOption == 'sortByCount') {
```

Comment:
i think we can move all the logic above this to inside this if block as we only need to calculate sortedLabels when `sortOption` is `sortByCount`